src/Microsoft.ML.Transforms/TensorflowTransform.cs

```diff
@@ -0,0 +1,601 @@
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Transforms;
+using Microsoft.ML.Transforms.TensorFlow;
+
+[assembly: LoadableClass(TensorflowTransform.Summary, typeof(TensorflowTransform), typeof(TensorflowTransform.Arguments), typeof(SignatureDataTransform),
+    TensorflowTransform.UserName, TensorflowTransform.ShortName)]
+
+[assembly: LoadableClass(TensorflowTransform.Summary, typeof(TensorflowTransform), null, typeof(SignatureLoadDataTransform),
+    TensorflowTransform.UserName, TensorflowTransform.LoaderSignature)]
+
+namespace Microsoft.ML.Transforms
+{
+    public sealed class TensorflowTransform : RowToRowMapperTransformBase
+    {
+        public sealed class Column : ManyToOneColumn
+        {
+            public static Column Parse(string str)
+            {
+                Contracts.AssertNonEmpty(str);
+
+                var res = new Column();
+                if (res.TryParse(str))
+                    return res;
+                return null;
+            }
+
+            public bool TryUnparse(StringBuilder sb)
+            {
+                Contracts.AssertValue(sb);
+                return TryUnparseCore(sb);
+            }
+        }
+
+        public sealed class Arguments : TransformInputBase
+        {
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "New column definition(s) (optional form: name:src)", ShortName = "col", SortOrder = 1)]
+            public Column[] Column;
+
+            [Argument(ArgumentType.Required, HelpText = "This is the frozen protobuf model file. Please see https://www.tensorflow.org/mobile/prepare_models for more detail(s).", ShortName = "ModelDir", SortOrder = 2)]
+            public string ModelFile;
+        }
+
+        private sealed class Bindings : ManyToOneColumnBindingsBase
+        {
+            public sealed class TFColInfo
+            {
+                public readonly string[] InputColNames;
+                public readonly TFShape[] TfShapes;
+                public readonly TFDataType[] TfTypes;
+
+                public TFColInfo(string[] inputColNames, TFShape[] tfShapes, TFDataType[] tfType)
+                {
+                    Contracts.AssertNonEmpty(tfShapes);
+                    Contracts.AssertNonEmpty(tfType);
+                    Contracts.Assert(tfType.Length == tfType.Length);
+
+                    InputColNames = inputColNames;
+                    TfShapes = tfShapes;
+                    TfTypes = tfType;
+                }
+            }
+
+            public readonly TFColInfo[] TfColInfo;
+            public readonly string[] OutputColNames;
+            public readonly ColumnType[] OutputCols;
+            public readonly TFDataType[] OutputTFTypes;
+
+            public Bindings(Column[] columns, ISchema schemaInput, TensorflowTransform parent)
```

Comment:
>Bindings [](start = 19, length = 8)

Change this class to have just one private ctor, and two public ctors/create methods that call the private one. This way if we make any changes to this class we won't need to remember to make them in two places.